qadence_protocols/mitigations/readout.py

```diff
@@ -116,48 +165,34 @@ def mitigation_minimization(
     corrected_counters: list[Counter] = []
 
     for sample in samples:
-        bitstring_length = 2**n_qubits
-        # List of bitstrings in lexicographical order.
-        ordered_bitstrings = [f"{i:0{n_qubits}b}" for i in range(bitstring_length)]
+        ordered_bitstrings = [bin(k)[2:].zfill(n_qubits) for k in range(2**n_qubits)]
         # Array of raw probabilites.
         p_raw = np.array([sample[bs] for bs in ordered_bitstrings]) / n_shots
 
         if optimization_type == ReadOutOptimization.CONSTRAINED:
-            # Initial random guess in [0,1].
-            p_corr0 = np.random.rand(bitstring_length)
-            # Stochasticity constraints.
-            normality_constraint = LinearConstraint(
-                np.ones(bitstring_length).astype(int), lb=1.0, ub=1.0
-            )
-            positivity_constraint = LinearConstraint(
-                np.eye(bitstring_length).astype(int), lb=0.0, ub=1.0
-            )
-            constraints = [normality_constraint, positivity_constraint]
-            # Minimize the corrected probabilities.
-            res = minimize(
-                corrected_probas, p_corr0, args=(noise_matrices, p_raw), constraints=constraints
-            )
-            p_corr = res.x
+            p_corr = constrained_inversion(noise_matrices, p_raw)
 
         elif optimization_type == ReadOutOptimization.MLE:
             noise_matrices_inv = list(map(matrix_inv, noise_matrices))
             # Compute corrected inverse using matrix inversion and run MLE.
             p_corr = mle_solve(tensor_rank_mult(noise_matrices_inv, p_raw))
+
+        elif optimization_type == ReadOutOptimization.MTHREE:
+            Confusion_matrix_subspace = subspace_kron(noise_matrices, p_raw.nonzero())
+            # GMRES is best suited for higher dimensional problems
```

Comment:
What is GMRES ?

Reply:
its an iterative solver for linear equations (Ax = b) suitable for high dimensional problems
changed the comment to "# GMRES (Generalized minimal residual) for linear equations in higher dimension"